lib/Service/AssistantService.php

```diff
@@ -66,50 +104,308 @@ public function sendNotification(TextProcessingTask|TextToImageTask $task, ?stri
+					$task = new TextProcessingTask(FreePromptTaskType::class, $input, $appId, $userId, $identifier);
+					$this->textProcessingManager->scheduleTask($task);
+					break;
+				}
+			default:
+				{
+					$input = $inputs['prompt'];
+					$task = new TextProcessingTask($type, $input, $appId, $userId, $identifier);
+					$this->textProcessingManager->scheduleTask($task);
+					break;
+				}
+		}
+
+		$assistantTask = $this->taskMapper->createTask($userId, $inputs, $task->getOutput(), time(), $task->getId(), $type, $appId, $task->getStatus(), Application::TASK_CATEGORY_TEXT_GEN, $identifier);
+
+		return $assistantTask;
+	}
+
+	/**
+	 * @param string $type
+	 * @param array<string> $inputs
+	 * @param string $appId
+	 * @param string|null $userId
+	 * @param string $identifier
+	 * @return Task
+	 * @throws PreConditionNotMetException
+	 * @throws \OCP\Db\Exception
+	 * @throws \Exception
+	 */
+	public function runOrScheduleTextProcessingTask(string $type, array $inputs, string $appId, ?string $userId, string $identifier): Task {
+		$inputs = $this->sanitizeInputs($type, $inputs);
+		switch ($type) {
+			case 'copywriter':
+				{
+					// Format the input prompt
+					$input = $this->formattedCopywriterPrompt($inputs['writingStyle'], $inputs['sourceMaterial']);
+					$task = new TextProcessingTask(FreePromptTaskType::class, $input, $appId, $userId, $identifier);
+					$this->textProcessingManager->runOrScheduleTask($task);
+					break;
+				}
+			default:
+				{
+					$input = $inputs['prompt'];
+					$task = new TextProcessingTask($type, $input, $appId, $userId, $identifier);
+					$this->textProcessingManager->runOrScheduleTask($task);
+					break;
+				}
+		}
+
+		$assistantTask = $this->taskMapper->createTask($userId, $inputs, $task->getOutput(), time(), $task->getId(), $type, $appId, $task->getStatus(), Application::TASK_CATEGORY_TEXT_GEN, $identifier);
+
+		return $assistantTask;
+	}
+
+	/**
+	 * Parse text from file (if parsing the file type is supported)
+	 * @param string $filePath
+	 * @param string $userId
+	 * @return string
+	 * @throws \Exception
+	 */
+	public function parseTextFromFile(string $filePath, string $userId): string {
+		try {
+			$userFolder = $this->storage->getUserFolder($userId);
+		} catch (\OC\User\NoUserException | NotPermittedException $e) {
+			throw new \Exception('Could not access user storage.');
+		}
+
+
+		try {
+			$mimeType = $userFolder->get($filePath)->getMimeType();
+		} catch (NotFoundException $e) {
+			throw new \Exception('File not found.');
+		}
+
+		try {
+			$file = $userFolder->get($filePath);
+			if ($file instanceof File) {
+				$contents = $file->getContent();
+			} else {
+				throw new \Exception('File is not a file.');
+			}
+		} catch (NotFoundException | LockedException | GenericFileException | NotPermittedException $e) {
+			throw new \Exception('File not found or could not be accessed.');
+		}
```

Comment:
You could first get the file, then check it's an instance of `OCP\Files\File`, then get the mime type. I mean let's avoid getting the file twice.

Reply:
Makes sense!